### PR TITLE
fix(smart-contracts): add tests for missing return statements in interface

### LINKED ISSUE #9877: Missing "returns" in ABI
Some of our functions are missing the `returns` statement in their interface even though we do actually return something :D
Example: 
`function publicLockVersions(address _impl) external view` should in fact be
`function publicLockVersions(address _impl) external view returns(uint16)`.

Let's fix them all if we can!

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds tests to detect missing return statements or other incoherent function declarations between interfaces and actual contarcts

# Issues

Fixes #9877
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Great idea!

**Review by julien51 — APPROVED** [review]
I guess we caught a few!

**Comment by clemsos** [comment]
> I guess we caught a few!

yep! Added the changes to the interfaces

File: smart-contracts/contracts/interfaces/IPublicLock.sol [hunk 1]
```diff
@@ -349,13 +351,14 @@ interface IPublicLock
 
   /**
    * @dev Determines how much of a refund a key owner would receive if they issued
-   * @param _keyOwner The key owner to get the refund value for.
+   * @param _tokenId the id of the token to get the refund value for.
    * a cancelAndRefund block.timestamp.
```
Comment (julien51) [inline smart-contracts/contracts/interfaces/IPublicLock.sol:355]:
```suggestion
```

File: smart-contracts/contracts/interfaces/IPublicLock.sol [hunk 2]
```diff
@@ -578,12 +580,16 @@ interface IPublicLock
   function hasRole(bytes32 role, address account) external view returns (bool);
 
   /**
-    * @notice An ERC-20 style transfer.
+    * @notice An ERC-20 style transfer for time.
```
Comment (julien51) [inline smart-contracts/contracts/interfaces/IPublicLock.sol:583]:
```suggestion
```

That is not an ERC20 transfer anymore...

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
LFG!

**Pull request merged**